### DISCUSSION TITLE
Everytime I open a js file,the treesitter error happened

### DISCUSSION THREAD

**Discussion by TwelveNight** [LazyVim/LazyVim#2153] [Q&A] [answered]
![image](https://github.com/LazyVim/LazyVim/assets/113519679/7066815e-8f15-4d8d-8d4c-66b7a1c0538c)

the checkhealth nvim-treesitter is here
![image](https://github.com/LazyVim/LazyVim/assets/113519679/33b68706-e893-429a-96d7-01384ae69923)

**Comment by dpetka2001** [comment]
How did you install Neovim? I had a similar problem/error with another language server and it ended up being caused by the Neovim installation itself (I was building from source, so I might have gone wrong in some step of the way). I have since transitioned to [bob](https://github.com/MordechaiHadad/bob) for easy way of installing Neovim and switching between stable and nightly.

**Reply by TwelveNight** [reply]
I`m a arch user,so I just use  pacman/yay,as you say I tried bob to swich between nightly and stable,but whether it is nightly or stable the error is the same for a js file

`nightly`
![image](https://github.com/LazyVim/LazyVim/assets/113519679/6dd5cf86-b84e-4afd-960c-4e7396860e9e)

`stable`
![image](https://github.com/LazyVim/LazyVim/assets/113519679/c3322c1f-d4c3-4287-87b0-52583ae64588)

**Reply by dpetka2001** [reply]
What does `which nvim` show?

**Reply by dpetka2001** [reply]
Did you uninstall previous Neovim from your package manager? `bob` encourages to not have another Neovim instance installed on your machine.

**Reply by TwelveNight** [reply]
I already remove the previous nvim package use yay -Rc neovim

this is `which nvim` show 
![image](https://github.com/LazyVim/LazyVim/assets/113519679/9f94c3f3-8c68-4959-966e-eee9efa92146)

**Reply by dpetka2001** [reply, marked as answer]
Do you happen to have `nvim-ts-rainbow` installed and configured via treesitter?

**Reply by TwelveNight** [reply]
yes,it is the problem!!
the config is here
```lua
return {
    "p00f/nvim-ts-rainbow",
    config = function()
      require("nvim-treesitter.configs").setup({
        -- ensure_installed = { "c", "lua", "vim", "vimdoc", "query", "go", "java" },

        -- highlight = { enable = true },
        -- indent = { enable = true },

        rainbow = {
          enable = true,
          extended_mode = true,
          max_file_lines = nil,
        },
      })
    end,
}
```
I configured this configuration a long time ago, but I no longer write the front end. I was still a very newbie at the time.

**Reply by dpetka2001** [reply]
Yes, as you can see in the github repo of the plugin, it has been deprecated and treesitter also stopped supporting modules for external plugins (if not entirely, definitely for this one, but the want to stop it completely as far as I know). If you'd like something similar, please take a look at [rainbow-delimiters](https://github.com/hiphish/rainbow-delimiters.nvim).
